Zenith/Controllers/EventsController.cs

```diff
@@ -22,7 +24,57 @@ public class EventsController : Controller
             var events = db.Events.Include(E => E.ActivityCategory);
             return View(await events.ToListAsync());
         }
+        public async Task<ActionResult> Schedule()
+        {
+            return View(getDic());
+        }
+        public ScheduleViewModel getDic()
+        {
+            DateTime startOfWeek = DateTime.Today.AddDays(
+                (int)CultureInfo.CurrentCulture.DateTimeFormat.FirstDayOfWeek -
+                (int)DateTime.Today.DayOfWeek);
+
+            string result = string.Join("," + Environment.NewLine, Enumerable
+              .Range(0, 7)
+              .Select(i => startOfWeek
+                 .AddDays(i)
+                 .ToString("dd-MMMM-yyyy")));
+            var weekDays = result.Split(',');
+            List<DateTime> dates = new List<DateTime>();
+            foreach(var w in weekDays)
+            {
+                dates.Add(Convert.ToDateTime(w));
+            }
+
+            //can't get this this to work; can't use datetimes in linq query, all online says is dbfunctions.truncatetime but still doesn't work
+            //var events = db.Events.Include(e => e.ActivityCategory).Where(e => dates.Any(d => DbFunctions.TruncateTime(d.Date) == e.CreationDate.Date)).ToList();
+
+            var events = db.Events.Include(e => e.ActivityCategory).ToList();
+            Dictionary<string, List<Event>> dic = new Dictionary<string, List<Event>>();
+            foreach (var d in dates)
+            {
+                dic.Add(GetDateString(d), new List<Event>());
+                foreach (var e in events)
+                {
+                    if(e.StartDateTime.Date == d.Date)
+                    {
+                        dic[GetDateString(d)].Add(e);
+                    }
+                }
+            }
+            
+
+            return new ScheduleViewModel()
+            {
+                DaysAndEvents = dic
+            };
 
+
+        }
+        public string GetDateString(DateTime date)
```

Comment:
You dont need this method. You can get this format of a DateTime object by just doing: \

`date.ToString("D")`